tests/test_integration/test_run_transfer/test_mssql.py

```diff
@@ -0,0 +1,296 @@
+        target_connection_id=mssql_connection.id,
+        source_params={
+            "type": "postgres",
+            "table_name": "public.source_table",
+        },
+        target_params={
+            "type": "mssql",
+            "table_name": "dbo.target_table",
+        },
+        queue_id=queue.id,
+    )
+    yield result
+    await session.delete(result)
+    await session.commit()
+
+
+@pytest_asyncio.fixture
+async def mssql_to_postgres(
+    session: AsyncSession,
+    group: Group,
+    queue: Queue,
+    mssql_for_conftest: MSSQL,
+    mssql_connection: Connection,
+    postgres_connection: Connection,
+):
+    result = await create_transfer(
+        session=session,
+        group_id=group.id,
+        name=f"mssql2postgres_{secrets.token_hex(5)}",
+        source_connection_id=mssql_connection.id,
+        target_connection_id=postgres_connection.id,
+        source_params={
+            "type": "mssql",
+            "table_name": "dbo.source_table",
+        },
+        target_params={
+            "type": "postgres",
+            "table_name": "public.target_table",
+        },
+        queue_id=queue.id,
+    )
+    yield result
+    await session.delete(result)
+    await session.commit()
+
+
+async def test_run_transfer_postgres_to_mssql(
+    client: AsyncClient,
+    group_owner: MockUser,
+    prepare_postgres,
+    prepare_mssql,
+    init_df: DataFrame,
+    postgres_to_mssql: Transfer,
+):
+    # Arrange
+    _, fill_with_data = prepare_postgres
+    fill_with_data(init_df)
+    mssql, _ = prepare_mssql
+
+    # Act
+    result = await client.post(
+        "v1/runs",
+        headers={"Authorization": f"Bearer {group_owner.token}"},
+        json={"transfer_id": postgres_to_mssql.id},
+    )
+    # Assert
+    assert result.status_code == 200
+
+    run_data = await get_run_on_end(
+        client=client,
+        run_id=result.json()["id"],
+        token=group_owner.token,
+    )
+    source_auth_data = run_data["transfer_dump"]["source_connection"]["auth_data"]
+    target_auth_data = run_data["transfer_dump"]["target_connection"]["auth_data"]
+
+    assert run_data["status"] == Status.FINISHED.value
+    assert source_auth_data["user"]
+    assert "password" not in source_auth_data
+    assert target_auth_data["user"]
+    assert "password" not in target_auth_data
+    reader = DBReader(
+        connection=mssql,
+        table="dbo.target_table",
+    )
+    df = reader.run()
+
+    # as spark rounds datetime up to milliseconds while writing to mssql: https://onetl.readthedocs.io/en/latest/connection/db_connection/mssql/types.html#id5
+    df = df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
+    init_df = init_df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
+
+    for field in init_df.schema:
+        df = df.withColumn(field.name, df[field.name].cast(field.dataType))
+
+    assert df.sort("ID").collect() == init_df.sort("ID").collect()
```

Comment:
Which case MSSQL uses by default, UPPERCASE or lowercase?

Reply:
Without column names normalization MSSQL uses lowercase

Reply:
In this case, shouldn't we compare like this?
```suggestion
    assert df.sort("id").collect() == init_df.sort("id").collect()
```